File: makefile/src/main/java/name/kropp/intellij/makefile/psi/impl/MakefileNamedElementImpl.java
package name.kropp.intellij.makefile.psi.impl;

import com.intellij.extapi.psi.StubBasedPsiElementBase;
import com.intellij.lang.ASTNode;
import com.intellij.psi.stubs.IStubElementType;
import name.kropp.intellij.makefile.psi.MakefileNamedElement;
import name.kropp.intellij.makefile.stub.MakefileTargetStubElement;
import org.jetbrains.annotations.NotNull;

public abstract class MakefileNamedElementImpl extends StubBasedPsiElementBase<MakefileTargetStubElement> implements MakefileNamedElement {
    public MakefileNamedElementImpl(@NotNull ASTNode node) {
        super(node);
    }

    public MakefileNamedElementImpl(@NotNull MakefileTargetStubElement stub, @NotNull IStubElementType nodeType) {
        super(stub, nodeType);
    }
}

File: makefile/src/main/kotlin/name/kropp/intellij/makefile/psi/MakefileNamedElementImpl.kt
package name.kropp.intellij.makefile.psi.impl

import com.intellij.extapi.psi.*
import com.intellij.lang.*
import com.intellij.psi.stubs.*
import name.kropp.intellij.makefile.psi.*
import name.kropp.intellij.makefile.stub.*

abstract class MakefileNamedElementImpl : StubBasedPsiElementBase<MakefileTargetStubElement>, MakefileNamedElement {
  constructor(node: ASTNode) : super(node)
  constructor(stub: MakefileTargetStubElement, nodeType: IStubElementType<*, *>) : super(stub, nodeType)

  override fun toString() = "MakefileTargetImpl(TARGET)"
}